src/hotspot/share/classfile/classFileParser.hpp

```diff
@@ -43,7 +43,7 @@ class ClassInstanceInfo;
 class CompressedLineNumberWriteStream;
 class ConstMethod;
 class FieldInfo;
-template <typename T>
+template<typename E, typename Index>
```

Comment:
```suggestion
template<typename E, typename Index = int>
```

Is it possible to reduce the changes by providing default parameters?

Reply:
Unfortunately, no. Forward decl.s may not re-define the default template argument, even though they are the same as the definition.